docs/usage/java.md

```diff
@@ -23,10 +23,52 @@ Renovate does not support:
 - Android projects that require extra configuration to run (e.g. setting the Android SDK)
 - Gradle versions prior to version 5.0.
 
+## Gradle Wrapper
+
+Renovate can update the [Gradle Wrapper](https://docs.gradle.org/current/userguide/gradle_wrapper.html) of a project.
+
+This includes the source declaration inside the `gradle/wrapper/gradle-wrapper.properties`
+as well as accompanied files such as `gradlew`, `gradlew.bat`, and `gradle/wrapper/gradle-wrapper.jar`.
+
 ### How It Works
 
-Renovate uses a plugin to search and extract versions from projects.
-Once the Gradle plugin has detected the dependencies, lookups and updating will be performed like usual with datasources and direct patching of files.
+Renovate extracts the Gradle Wrapper version used from the `distributionUrl` inside the `gradle-wrapper.properties`.
+Once the version is determined, Renovate will look for newer versions from the `gradle-version` datasource.
+Renovate will then invoke the Gradle Wrapper to update itself,
+[as recommended by Gradle](https://docs.gradle.org/current/userguide/gradle_wrapper.html#sec:upgrading_wrapper).
+
+For the extraction to work, the `distributionUrl` must point to
+a file of type .zip,
+which includes the version in its name, and
+defines one of the official distribution types (bin, all).
+
+### Support for mirrors and custom distributions
+
+As Renovate takes the `distributionUrl` defined inside the `gradle-wrapper.properties` as basis for its update,
+source declarations other than to the official Gradle Wrapper are supported.
+
+This can be used for hosting the official distributions with a proxy server, an offline mirror
+or even providing a custom distribution of the Gradle Wrapper,
+e.g. to provide a company-wide base configuration for all Gradle projects.
+
+However, the `gradle-version` datasource is used to determine available versions.
+In case the available versions at the defined source differ
+from those available from Gradle
+or the [default datasource](https://services.gradle.org/versions/all) cannot be reached, e.g. due to network restrictions,
+the datasource may be reconfigured via a `packageRule`.
```

Comment:
```suggestion
In case the available versions at the defined source differ from those available from Gradle or the [default datasource](https://services.gradle.org/versions/all) cannot be reached, e.g. due to network restrictions, the datasource may be reconfigured via a `packageRule`:
```

Use one sentence per line.
Add a `:` at the end of the sentence, as we want people to read the package rule example.